File: cpp/luogu/old/P1093.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\p1093.cpp
 *@data      2024/01/17 20:11:59
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

struct sdt {
    int num, cn, tot;
}qwq[527];

bool cmp(sdt a, sdt b) {
    if(a.tot != b.tot)return a.tot > b.tot;
    else {
        if(a.cn != b.cn)return a.cn > b.cn;
        else return a.num < b.num;
    }
}
void solve() {
    int n;
    cin >> n;
    for(int i = 1;i <= n;++i) {
        int c, m, e;
        qwq[i].num = i;
        cin >> qwq[i].cn >> m >> e;
        qwq[i].tot = qwq[i].cn + m + e;
    }
    sort(qwq + 1, qwq + 1 + n, cmp);
    for(int i = 1;i <= 5;++i)cout << qwq[i].num << ' ' << qwq[i].tot << '\n';
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/p1093.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1093.rs
//@data      2024/01/17 23:47:58
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
struct Sc {
    id: usize,
    cn: usize,
    tot: usize,
}
fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let n: usize = cin().parse().unwrap();
    let mut qwq: Vec<Sc> = vec![];
    for i in 1..=n {
        let (a, b, c) = cin()
            .split_whitespace()
            .fold((0, 0, 0), |x, y| (x.1, x.2, y.parse::<usize>().unwrap()));
        qwq.push(Sc {
            id: i,
            cn: a,
            tot: a + b + c,
        });
    }
    qwq.sort_unstable_by(|b, a| {
        if a.tot != b.tot {
            a.tot.cmp(&b.tot)
        } else if a.cn != b.cn {
            a.cn.cmp(&b.cn)
        } else {
            b.id.cmp(&a.id)
        }
    });
    qwq[0..5]
        .iter()
        .for_each(|x| println!("{} {}", x.id, x.tot));
}
